src/dataservice/data-service-dao-file/src/main/java/com/redhat/depdraw/dataservice/dao/file/DiagramDaoImpl.java

```diff
@@ -0,0 +1,83 @@
+package com.redhat.depdraw.dataservice.dao.file;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.OpenOption;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.redhat.depdraw.dataservice.dao.api.DiagramDao;
+import com.redhat.depdraw.dataservice.dao.model.Diagram;
+
+@ApplicationScoped
+public class DiagramDaoImpl implements DiagramDao {
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    @Override
+    public Diagram create(Diagram d) {
+        return createInternal(d, StandardOpenOption.CREATE_NEW);
+    }
+
+    private Diagram createInternal(Diagram d, OpenOption... openOptions) {
+        try {
+            final String s = objectMapper.writeValueAsString(d);
+            final String pathString = FileUtil.DIAGRAM_FILES_DIR + d.getUuid() + "/";
+            Path path = Path.of(pathString);
+
+            Files.createDirectories(path);
+
+            Path fileName = Path.of(pathString + FileUtil.DIAGRAM_FILE_NAME);
+
+            // Writing into the file
+            Files.writeString(fileName, s, openOptions);
+        } catch (IOException e) {
+            return d;
+        }
+
+        return d;
+    }
+
+    @Override
+    public Diagram getDiagramById(String diagramId) {
+        try {
+            Path fileName = Path.of(FileUtil.DIAGRAM_FILES_DIR + diagramId
+                    + "/" + FileUtil.DIAGRAM_FILE_NAME);
```

Comment:
what about separating the logic to calculate the path in a reusable method/utility?